src/applications/coronavirus-chatbot/createCoronavirusChatbot.js

```diff
@@ -1,29 +1,51 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import initializeChatbot from './index';
 import recordEvent from '../../platform/monitoring/record-event';
 import { GA_PREFIX } from './utils';
 
-export default (_store, widgetType) => {
+export default (store, widgetType) => {
   // Derive the element to render our widget.
   const root = document.querySelector(`[data-widget-type="${widgetType}"]`);
 
   // Escape early if no element was found.
   if (!root) {
     return;
   }
-
-  import(/* webpackChunkName: "chatbot" */ './index')
+  // webpackChunkName: "chatbot"
+  import('./chatbot-entry')
     .then(module => {
-      const initializeChatbot = module.default;
-      initializeChatbot(root);
+      const { CoronavirusChatbot } = module.default;
+      initializeChatbot()
+        .then(webchatOptions => {
+          ReactDOM.render(
+            <Provider store={store}>
```

Comment:
I am not up to speed on React by any means, so this might be an anti-pattern if so disregard(or please inform me otherwise). In Vue we typically try to separate templates from logic even programmatic templates into their own render statements, this allows for less dense and theoretically more readable code. This whole file seems to add to the code density, without promoting readability, and I would ask maybe we refactor it(if this is a good practice ) so future us can work more quickly with it.

Reply:
@katieho-va sorry this is one of my other githubs :(